daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1293,6 +1291,7 @@ private[lf] object SBuiltin {
     override private[speedy] def execute(args: util.ArrayList[SValue], machine: Machine): Unit = {
       checkToken(args, 0)
       // $ugettime :: Token -> Timestamp
+      machine.withOnLedger(NameOf.qualifiedNameOfCurrentFunc)(_.dependsOnTime = true)
```

Comment:
I’m confused, did this work before?

Reply:
Oh I see, we set it in the engine instead of in speedy so it did work.

Reply:
tried to fix your comment from 21 June (See unresolved conversation above) 
> Unrelated to this PR but this seems weird. Why don’t we modify this in speedy?

Reply:
seems like past me was smarter than me :)